backend/x86_proc.ml

```diff
@@ -15,6 +15,70 @@
 
 open X86_ast
 
+module StringTbl = Hashtbl.Make (struct
+    type t = string
+
+    let equal = String.equal
+
+    let hash = Hashtbl.hash
+  end)
+
+module SectionName = struct
+  type t =
+    { name: string list;
+      name_str: string;
+      flags : string option;
+      args: string list
+    }
+
+  let equal t1 t2 =
+    List.equal (String.equal) t1.name t2.name
+
+  let hash t = Hashtbl.hash t.name
+
+  let compare t1 t2 = List.compare (String.compare) t1.name t2.name
+
+  let make name flags args =
+    { name; name_str = String.concat "," name; flags; args; }
+
+  let from_name name = { name = [name]; name_str = name; flags = None; args = [] }
+
+  let name t = t.name_str
+
+  let flags t =
+    let parse_flags flags =
+      let rec inner acc = function
+        | Seq.Nil -> acc
+        | Seq.Cons (c, tl) ->
+          let flag =
+            match c with
+            (* CR mcollins - modify types to avoid string comparisons *)
+            | 'a' -> 0x2L
```

Comment:
Can you please add a comment somewhere pointing to the document where these encodings (and the ones in `internal_assembler` subdirectory) are coming from (ELF ABI?).